src/WWT.PlateFiles/StreamSlice.cs

```diff
@@ -69,15 +69,18 @@ public StreamSlice(Stream baseStream, long offset, long length)
         }
 
         public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
-            => ReadInternalAsync(buffer, offset, count, true, cancellationToken).AsTask();
+            => ReadAsync(buffer.AsMemory(offset, count), cancellationToken).AsTask();
 
         public override int Read(byte[] buffer, int offset, int count)
-            => ReadInternalAsync(buffer, offset, count, isAsync: false, default).Result;
+            => Read(buffer.AsSpan(offset, count));
 
-        /// <summary>
-        /// A read implementation that can be either async or synchronous to reduce duplication of code. The synchronous pathway will always be completed.
-        /// </summary>
-        private async ValueTask<int> ReadInternalAsync(byte[] buffer, int offset, int count, bool isAsync, CancellationToken cancellationToken)
+        public override int ReadByte()
+        {
+            Span<byte> buffer = stackalloc byte[1];
+            return Read(buffer) == 0 ? -1 : buffer[0];
```

Comment:
What is this for? Reading random bytes from the stack frame? I'm not sure if we can depend on the precise location `buffer` is allocated from not changing depending on the context this is called from.

Reply:
The default implementation of `ReadByte()` will allocate a 1 entry array for each read. Instead, this will allocate it on the stack and use the `Read(Span<byte>)` overload to fill it. If it didn't read, it would return `-1`, while if it did, the value is returned.

Reply:
Makes sense. Thanks for clarifying.